torch_xla/csrc/tensor.cpp

```diff
@@ -1835,6 +1847,24 @@ Device XLATensor::CommonDeviceForTensors(
   return device;
 }
 
+std::vector<XLATensor> XLATensor::WrapIndicesOnce(
+    const XLATensor& input,
+    tensorflow::gtl::ArraySlice<const XLATensor> indices) {
+  std::vector<XLATensor> canonical_indices;
+  XLA_CHECK_LE(indices.size(), input.shape().get().rank());
+  for (size_t dim_idx = 0; dim_idx < indices.size(); ++dim_idx) {
+    const XLATensor& dim_index = indices[dim_idx];
+    int64_t dim_size = input.shape().get().dimensions(dim_idx);
+    XLATensor wrapped_dim_index =
+        Create(dim_index.GetIrValue() +
+                   ir::ops::ScalarOp(at::Scalar(dim_size), dim_index.shape()),
+               input.GetDevice());
+    XLATensor wrap_cond = lt(indices[dim_idx], at::Scalar(int64_t(0)));
```

Comment:
You can just call XLATensor::lt() and XLATensor::where() ... they are static anyway.

Reply:
Ok, but they take tensors. So we'd have a situation in which we pass `XLATensor` inputs and get back an `ir::Value`, then? That also means the `IndexOp` would be subject to the same weirdness of receiving `XLATensor` inputs.

Reply:
I don't understand. You are handling XLATensors here ... and the XLATensor::lt() and XLATensor::where() APIs are both public and static.
What is the problem?
You mean that XLATensor will use inde_op.cpp/h, and vice versa?
As long as include files don't cross-fire to each other, it should be OK.
Not pretty, but OK.

Reply:
Ok, let's recap. If we move `WrapIndicesOnce` out of `XLATensor`, we lose access to `XLATensor::Create` (since it's private). This means we can't create `wrapped_dim_index` anymore as an `XLATensor`, which in turn prevents us from using `where`.

Reply:
I think creating an XLATensor from an IR Node, can be made public. I do not see big damage there.
Alternatively, create an at::Tensor from Scalar+sizes, and use the public ctor.

Reply:
Alright, I'll make that `Create` overload public and be done with it.

Reply:
I think it might get useful in other cases where we have some infrastructure that do not strictly belong to XLATensor, but still wanting to create them based on IR Nodes.

Reply:
Sounds good, done.